onnx/version_converter/convert.h

```diff
@@ -0,0 +1,160 @@
+#include "onnx/version_converter/adapters/NoPreviousVersionAdapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class DefaultVersionConverter : public BaseVersionConverter {
+  private:
+    bool DEBUG = false;
+
+    bool searchOpDomainMap(const std::unordered_map<std::string, std::map<
+      int64_t, const OpSchema*>>& op_domain_map, int64_t curr_version) const {
+      const auto version_it = op_domain_map.find("");
+      return version_it != op_domain_map.end() &&
+          version_it->second.find(curr_version) !=
+          version_it->second.end();
+    }
+
+  public:
+    DefaultVersionConverter() {
+      // Register adapters to the version converter
+      registerAdapter(make_unique<NoPreviousVersionAdapter>("Cos",
+        OpSetID(7), OpSetID(6)));
+    }
+
+    ModelProto convert_version(
+        const ModelProto& mp_in,
+        const OpSetID& initial_version,
+        const OpSetID& target_version) const override {
+      const std::string initial_domain = initial_version.domain();
+      const std::string target_domain = target_version.domain();
+
+      ONNX_ASSERTM((initial_domain == "" || initial_domain == "ai.onnx") &&
+        (target_domain == "" || target_domain == "ai.onnx"),
+          "Warning: default onnx version converter can only convert "
+          " between default domain opset versions ('' or 'ai.onnx')\n"
+          "Provided initial_domain: %s, provided target_domain: %s",
+          initial_domain.c_str(), target_domain.c_str());
+
+      ONNX_ASSERTM(initial_domain == target_domain,
+        "initial_version and target_version must have the same domains");
+      for (auto it = mp_in.opset_import().begin(); it != mp_in.opset_import()
+          .end(); ++it) {
+        if (it->domain() == initial_version.domain()) {
+          ONNX_ASSERTM(initial_version.version() == it->version(),
+              "initial_version does not reflect current state of model");
+        }
+      }
+
+      std::shared_ptr<Graph> g(ImportModelProto(mp_in));
+      ONNX_ASSERTM(g.get() != nullptr,
+        "Warning: onnx version converter is unable to parse input model. "
+        "(The IR version of the ONNX model may be too old.)");
+
+      // TODO: Move to Inter-Domain Converter
+      // Get initial model versions
+      // std::vector<OpSetID> initial_versions = g->opset_versions_mutable();
+
+      // No conversion necessary if Model has single, equivalent opset version
+      // if (initial_versions.size() == 1 && initial_versions[0].version ==
+      //    target_version.version && initial_versions[0].domain ==
+      //    target_version.domain) {
+      //  return mp_in;
+      // }
+
+      // Check if target_version is valid
+      const std::unordered_map<std::string, std::pair<int, int>>& versions_map = OpSchemaRegistry::DomainToVersionRange::Instance().Map();
+      const std::string search_domain = target_version.domain() == "ai.onnx" ? "" : target_version.domain();
```

Comment:
directly set ""?

Reply:
actually, please confirm what string ("" or "ai.onnx") we use in `versions_map`

Reply:
We always use "" in versions_map, so this builds in compatibility for users providing "ai.onnx" in reference to the same domain.